packages/augur-core/source/contracts/CashSender.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity 0.5.10;
+
+import 'ROOT/ICash.sol';
+import 'ROOT/external/IDaiVat.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract CashSender {
+    using SafeMathUint256 for uint256;
+
+    ICash public cash;
+    IDaiVat public vat;
+
+    uint constant RAY = 10 ** 27;
+
+    function initializeCashSender(address _vat, address _cash) internal {
+        vat = IDaiVat(_vat);
+        cash = ICash(_cash);
+    }
+
+    function cashTransfer(address _to, uint256 _amount) internal {
+        address _from = address(this);
+        if (vat.live() == 0) {
+            _amount = shutdownTransfer(_from, _to, _amount);
+            if (_amount == 0) {
+                return;
+            }
+        }
+        require(cash.transfer(_to, _amount));
```

Comment:
In other places, there is protection in place to make sure things function even if cash balances are less. I haven't fully worked it out, but the fact that Augur overpays on withdraw:

```
if (_sDaiAmount.mul(_chi) < _amount.mul(RAY)) {
            _sDaiAmount += 1;	            _sDaiAmount += 1;
        }	        
}
```

and during a shutdown, DSR = ONE (so not interest accruing), we could end up every-so-slightly under?

Reply:
I don't think in any of the places this is used there is an expectation or an allowance that the balance is less actually.

The referenced bit for DSR handling should only come into effect when interest has accrued and would generate a rounding error. We do not allow an overdraw in that code from the pot so if for some reason the rounding error case is triggered but there is insufficient money in the pot (not sure how it would happen) we'd just fail there.

Reply:
Yes i think you are correct. the only time it can round down is if it has already accrued interest since sDaiAmount * chi will not necessarily be divisible by the deposited vDai amount